### PR TITLE
Fix HealthCheck errors when editor disabled

### PR DESCRIPTION
**Pull request by hardillb**
fixes FlowFuse/flowfuse#4393

## Description

When the editor is disabled the healthcheck gets a 404, this stops it being an error

## Related Issue(s)

 FlowFuse/flowfuse#4393

## Checklist

 - [x] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [ ] Suitable unit/system level tests have been added and they pass 
 - [ ] Documentation has been updated
    - [ ] Upgrade instructions
    - [ ] Configuration details
    - [ ] Concepts
 - [ ] Changes `flowforge.yml`?
    - [ ] Issue/PR raised on `FlowFuse/helm` to update ConfigMap Template
    - [ ] Issue/PR raised on `FlowFuse/CloudProject` to update values for Staging/Production

## Labels

 - [ ] Includes a DB migration? -> add the `area:migration` label

### CODE REVIEW

File: lib/launcher.js [hunk 1]
```diff
@@ -424,6 +424,9 @@ class Launcher {
                     timeout: { request: this.healthCheckInterval - 500 },
                     retry: { limit: 0 }
                 }
+                if (this.settings.settings.disableEditor) {
+                    opts.throwHttpErrors = false
+                }
```
Comment (Steve-Mcl) [inline lib/launcher.js:429]: Should we not simply be disabling health poll or skip out of it if editor is disabled Ben?

I mean this is gonna be polling a non-existent endpoint and getting a 404 (I assume) so should we not just skip out of the health check?

Comment (hardillb) [reply lib/launcher.js:429]: No, this is still valid as the request will timeout if the NR instance is overworked before returning a 404

Comment (hardillb) [reply lib/launcher.js:429]: I'm nearly tempted to not if block this and just set it on the options for all health check requests

Comment (Steve-Mcl) [reply lib/launcher.js:429]: Lets go with what you tested for now Ben.

**Review by Steve-Mcl — APPROVED** [review]
I know you were in 2 minds about the health check options so while I approve the current fix, I will leave it for you to merge ben (in case you decide it would be better approach)

**Pull request merged**